### ISSUE TITLE
It wont work on my Mac

### ISSUE THREAD

**Issue by felippenardi** [Morhaus/dispatch-proxy#42]
The internet is not being distributed to other networks. I've set my wifi as the priority network, running a 1mbps 3G. At the same time I had a second priority ethernet cable connected with a 30mbps internet.

Without dispatch, the internet is 1mbps.
With `dispatch start`, the internet is 1mpbs
With dispatch targeted only to the 30mbps network, the internet is 1mpbs.

I've hit "apply" when setting the socks connection.
I've already tried http only.
I've installed npm with sudo.
I've tried running `dispatch` command with and without sudo

When running the `--debug` flag, it will seem to be dispatching accordingly to what I've set, but it does actually not.

**Comment by felippenardi** [comment]
I ensured that the connections are really going through the proxy with the following tests:

!!! Without dispatch
![wihtout-dispatch](https://cloud.githubusercontent.com/assets/1354492/7076335/0c50dd62-dedf-11e4-8845-e36742ede8c7.png)

You can see that Little Snitch register Chrome network activity. 

!!! With dispatch
![with-dispatch](https://cloud.githubusercontent.com/assets/1354492/7076334/0c0d3346-dedf-11e4-8f37-486c8455165c.png)

You can see that Little Snitch register iTerm Node network activity. And If I kill `dispatch`, I can not navigate.

---

Still, the internet velocity I get is always equivalent to the primary network. And I can tell it because it a 5mbps vs 30mbps speed, so it is easy to see. On my test, I'm using the 5mpbs network as primary.

Here is the logs:

```
$ dispatch start --debug
SOCKS server started on localhost:1080
Dispatching to addresses 192.168.0.10@1, 192.168.0.100@1
   request [39373e] api.4titude.net:443
  dispatch [39373e] 192.168.0.10
   connect [39373e] api.4titude.net:443
       end [39373e] clientConnection
   request [65649d] dl.bondwidth.org:80
  dispatch [65649d] 192.168.0.100
       end [39373e] serverConnection
   connect [65649d] dl.bondwidth.org:80
   request [b73fd3] api.4titude.net:443
  dispatch [b73fd3] 192.168.0.10
   connect [b73fd3] api.4titude.net:443
       end [b73fd3] clientConnection
   request [3a8d8e] api.4titude.net:443
  dispatch [3a8d8e] 192.168.0.10
   request [e50a0b] api.4titude.net:443
  dispatch [e50a0b] 192.168.0.100
   connect [3a8d8e] api.4titude.net:443
   connect [e50a0b] api.4titude.net:443
   request [0bc9bb] talkgadget.google.com:443
  dispatch [0bc9bb] 192.168.0.10
   connect [0bc9bb] talkgadget.google.com:443
       end [3a8d8e] serverConnection
       end [3a8d8e] clientConnection
       end [e50a0b] clientConnection
       end [65649d] clientConnection
       end [e50a0b] serverConnection
       end [65649d] serverConnection
   request [ed0104] github.com:443
  dispatch [ed0104] 192.168.0.100
   connect [ed0104] github.com:443
   request [a2b611] assets-cdn.github.com:443
  dispatch [a2b611] 192.168.0.100
   request [66afd0] assets-cdn.github.com:443
  dispatch [66afd0] 192.168.0.10
   request [f983d1] avatars3.githubusercontent.com:443
  dispatch [f983d1] 192.168.0.100
   request [fb683d] assets-cdn.github.com:443
  dispatch [fb683d] 192.168.0.10
   connect [66afd0] assets-cdn.github.com:443
   connect [fb683d] assets-cdn.github.com:443
   connect [f983d1] avatars3.githubusercontent.com:443
   connect [a2b611] assets-cdn.github.com:443
   request [8b0887] avatars1.githubusercontent.com:443
  dispatch [8b0887] 192.168.0.100
   request [164edc] avatars1.githubusercontent.com:443
  dispatch [164edc] 192.168.0.10
   request [7c8cd6] assets-cdn.github.com:443
  dispatch [7c8cd6] 192.168.0.100
   request [20646e] assets-cdn.github.com:443
  dispatch [20646e] 192.168.0.10
   connect [8b0887] avatars1.githubusercontent.com:443
   connect [164edc] avatars1.githubusercontent.com:443
   connect [20646e] assets-cdn.github.com:443
   request [28a297] collector-cdn.github.com:443
  dispatch [28a297] 192.168.0.100
```

**Comment by felippenardi** [comment]
@Morhaus Does that help?

**Comment by Morhaus** [comment]
@felippenardi That's really strange. If one connection is inactive, only half of the connections should go through.

What version of NodeJS are you using?

**Comment by Morhaus** [comment]
Also, keep in mind that one connection can only be dispatched to one network interface, so to really test your composite download speed you need to run several downloads at the same time.

**Comment by felippenardi** [comment]
My node is `v0.10.32`.

So running speed test would not be a good way? Because that is the only thing I'm trying out until now. I'll try something different :)

**Comment by felippenardi** [comment]
But actually... I think that is not the case. Because take look on the scenario:

I have two network interfaces:
- Wifi (manually set to primary): 5mpbs
- Ethernet: 30mbps

Both are connected, and the Wifi is primary. I turn on dispatch pointing ONLY to the Ethernet. My speed test will always max to 5mpbs.

**Comment by Morhaus** [comment]
No, speedtest is not a good way to test the effect of dispatch-proxy, as it will only create one connection to the distant server, thus only using one network interface.

It looks like the connection will go through wifi no matter what. To find out if that is the case, you should monitor the activity of both network interfaces when testing out a configuration.

I use iStat Menu.
![image](https://cloud.githubusercontent.com/assets/1621758/7105044/4be9d28a-e105-11e4-9616-05a8299236ee.png)

**Comment by felippenardi** [comment]
@Morhaus Perfect suggestion! Let me try that :)

**Comment by felippenardi** [comment]
@Morhaus Thank you so much! I finally got it working :)

I guess one of the problems was that I was setting only the primary network to use SOCKS. I set both now, and with your suggestion I could see that both network interfaces are being used.

I even managed to run two speed tests at the same time and get the exact result I expect for each network interface :)

![screen shot 2015-04-12 at 2 18 38 pm](https://cloud.githubusercontent.com/assets/1354492/7106647/01035506-e11f-11e4-9b42-48b2df35bfe0.png)
![screen shot 2015-04-12 at 2 15 47 pm](https://cloud.githubusercontent.com/assets/1354492/7106648/0383c4aa-e11f-11e4-974e-db2292b9c752.png)
![screen shot 2015-04-12 at 2 15 41 pm](https://cloud.githubusercontent.com/assets/1354492/7106649/06d76562-e11f-11e4-9a7c-6fa6735daa28.png)
